vnext/ReactWindowsCore/DevSettings.h

```diff
@@ -45,10 +46,16 @@ enum class JSIEngineOverride : int32_t {
   Last = V8Lite
 };
 
+struct ScriptMetadata {
+  uint64_t scriptVersion;
+  std::string bytecodeFilename;
+};
+
+using ScriptUrlMetadataMap = std::map<std::string, ScriptMetadata>;
```

Comment:
>map [](start = 34, length = 3)

from what i remember unordered_map has better perf, please double check

Reply:
I know MSVC has some issues with unordered_map, and unordered_map is still node based instead of using open addressing. IIRC std::map performed better for small sizes, but now I do want to check this again. Eventually when std::flat_map arrives, that would be best since a BST or hash table is way overkill for the number of things we pass in.

---
In reply to: [342724287](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342724287) [](ancestors = 342724287)

Reply:
We only have four to five bundles max right now so honestly I don't think this matters much. Plus it is going away so.

---
In reply to: [342860605](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342860605) [](ancestors = 342860605,342724287)

Reply:
Just out of curiosity I tried to find an answer, and it seems like a tossup as to which is better in the general case. One factor is hash cost for a long string vs comparions, but a lot of paths would have a shared prefix so we wouldn't exit a string comparison super early anyway. Intuitively I would still think a BST is faster since with a low number of entries we wouldn't really do a lot of pointer chasing, but as Yichen mentioned, we're kind of splitting hairs when we have such a low number of items.

I know at least one person with stronger opinions on the issue, but cannot find some of the evidence sited when he wrote this. I think this was guidance for DOCS?

```
Avoid std unordered_map
Short reason: it's harder to use correctly, and it has worse performance vs. std::map in most cases.

Detailed reply from STL (an engineer on VC libraries team):

map offers guaranteed log perf, unordered_map is subject to the quality of the hash and is vulnerable to DOS attacks unless you specifically choose a resistant hash (and I am not certain of the current best choice; many candidates have been later identified as suboptimal).
Writing less-than is easier than writing equal-to and a high quality hash for user-defined types.
Both map and unordered_map are pigs in terms of memory usage, but unordered_map is probably worse.
They are definitely both pigs in terms of locality, since the Standard requires node-based implementations.
It's hard to screw up a map, but there's a lot of variance in unordered_map implementations. Our implementation is very old (shared with the non-Standard hash_map) and was 2x worse than others when I last checked. It needs a representation overhaul in a major version, which I have taken a step towards by deprecating hash_map in Dev14 for removal in the next major version.
```

---
In reply to: [342862514](https://github.com/microsoft/react-native-windows/pull/3591#discussion_r342862514) [](ancestors = 342862514,342860605,342724287)